src/pal/src/exception/seh-unwind.cpp

```diff
@@ -73,17 +74,17 @@ struct __cxa_exception
 // Virtual Unwinding
 //----------------------------------------------------------------------
 
-static void WinContextToUnwindContext(CONTEXT *winContext, unw_context_t *unwContext)
+static void WinContextToUnwindContext(CONTEXT *winContext, unw_cursor_t *cursor)
 {
+    int st;
 #if defined(_AMD64_)
-    unwContext->uc_mcontext.gregs[REG_RIP] = winContext->Rip;
-    unwContext->uc_mcontext.gregs[REG_RSP] = winContext->Rsp;
-    unwContext->uc_mcontext.gregs[REG_RBP] = winContext->Rbp;
-    unwContext->uc_mcontext.gregs[REG_RBX] = winContext->Rbx;
-    unwContext->uc_mcontext.gregs[REG_R12] = winContext->R12;
-    unwContext->uc_mcontext.gregs[REG_R13] = winContext->R13;
-    unwContext->uc_mcontext.gregs[REG_R14] = winContext->R14;
-    unwContext->uc_mcontext.gregs[REG_R15] = winContext->R15;
+    st = unw_set_reg(cursor, UNW_REG_IP, winContext->Rip);
```

Comment:
I still don't see how this could work when we don't set the SP. My previous comment on this got somehow erased by github. I've said that for a single IP, there can be multiple frames with the same IP e.g. for recursive functions.